torchvision/prototype/models/detection/faster_rcnn.py

```diff
@@ -147,11 +147,25 @@ def _fasterrcnn_mobilenet_v3_large_fpn(
     return model
 
 
+# Unfortunately, the addition and removal of parameters here is BC breaking for
+# users that are using positional arguments.
+#
+# It might be possible to preserve BC by allowing:
+# weights to be a bool - it maps to pretrained
+# weights_backbone to be a bool - it maps to progress
+# progress to be an int - it maps to num_classes
+# num_classes to be a bool: it maps to pretrained_backbones
+# And to raise a warning in all these cases.
+#
+# That will complicate the code though, we should only do this if we are 100%
+# sure to remove these deprecated behaviours in 1 or 2 versions.
+# Also, this only works because the types of the new vs old parameters order
+# don't overlap.
```

Comment:
It's a tough decision...
I feel like BC is probably *the* most important contract that we have with our users. I would assume that if there are occurrences of positional arguments in fbcode, we can expect that there are a whole lot of them out in the world.

I wouldn't want to delay the new API either though. My preferred course of action would be to preserve BC. For cases like `resnet18` it should be fairly easy to keep BC and still do the clean up (as per the suggestion below). For cases like here or `keypointrcnn_resnet50_fpn`, that probably means we can't "clean up" the API - but it's a fair price to pay I think.

----

Then, irrespective of this new API, and for the upcoming release, I would suggest to start allowing the the clean ups by e.g. forcing users to use keyword-only arguments (this can be done smoothly with stuff like https://scikit-learn-enhancement-proposals.readthedocs.io/en/latest/slep009/proposal.html#deprecation-path). I think our future-selves will thank us in 2-3 versions.

Reply:
I think the most important one to keep is just `resnet50(True)` (i.e., for the `pretrained` argument being the first one), the others I think we don't need to worry about positional arguments.

Reply:
Unless I'm missing something, it's possible to do fully-BC across all builders by moving the parameters on the original position and applying workaround like the one proposed by @NicolasHug here:
https://github.com/pytorch/vision/blob/500f826e97b20848863574523227644b40a6bf69/torchvision/prototype/models/resnet.py#L255-L261

We can do the same for `weights_backbone`. 

I think from both your feedback it's clear that maintaining compatibility for things like `resnet50(True)` is important and perhaps we should aim for full-BC until we decide to clean up the API on a separate coordinated initiative.

Reply:
Addressed at PR #4948 commit 6030338780decdf24e36b5045b960d05463e6766